### DISCUSSION TITLE
How to perform an edge flip

### DISCUSSION THREAD

**Discussion by Alan-Leo-Wong** [BrunoLevy/geogram#197] [Q&A] [answered]
Hi, could you tell me how to perform an edge flip for a given mesh?

I’m currently working on a task where I’ve used a mesh's vertices as seeds to compute a restricted Delaunay triangulation in surfacic mode. I want to flip some of the edges between vertices in the triangulation to restore the connectivity structure of the original mesh. However, I seem to be unable to find any relevant functions for performing edge flips.

**Comment by BrunoLevy** [comment, marked as answer]
Hello, there is no function to perform an edge flip in geogram, but you can easily implement it, by using `mesh.facet_corners.set_vertex()` and `mesh.facet_corners.set_opposite_facet()`. In the end it will be very similar to
[this code](https://github.com/BrunoLevy/geogram/blob/0b5f9edca57eb46b00c9eb28764dfcef6e436b54/src/lib/geogram/delaunay/CDT_2d.cpp#L1057) in the constrained 2D triangulation but beware the [indexing is different](https://github.com/BrunoLevy/geogram/wiki/Mesh#triangulated-and-polygonal-meshes).
